### PR TITLE
Use GitHub Actions for CI

### LINKED ISSUE #4130: Try GitHub Actions for CI
We currently use Travis to run automated builds and tests for p5 development on GitHub. I've tried out an experiment in using the new [GitHub Actions](https://github.com/features/actions), which is basically a CI (and other automation) toolkit for GitHub.

My goal in trying this is to make the process of finding and understanding automated build and testing easier for p5js contributors. There's not really a difference in functionality between GitHub Actions and Travis for us (as far as I can tell).

### PR DESCRIPTION
**Pull request by outofambit**
closes #4130 

## Description

* adds GitHub Actions configuration for automated building and testing
* split testing process into subtasks so they can be run separately by CI
* remove travis config

## Demo

[Check out this test PR to see what it looks like!](https://github.com/outofambit/p5.js/pull/5)
GitHub Actions CI interface

## To Do

if people are generally 👍 to this, i will
- [x] update the docs
- [x] remove a few last references to travis from the repo
- [x] re-add test coverage reporting to CI

### CODE REVIEW

**Comment by lmccart** [comment]
looks good to me!

File: .github/workflows/ci-lint-examples.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+name: Linting Examples
+
+on: [push]
```
Comment (gruselhaus) [inline .github/workflows/ci-lint-examples.yml]:
```suggestion
on: [push, pull_request]
```
If the set event is `push` the workflow will be triggered by push events in the same repo. As a result a pull request from a fork will not trigger this workflow.

See https://help.github.com/en/actions/automating-your-workflow-with-github-actions/events-that-trigger-workflows#about-workflow-events

File: .github/workflows/ci-lint.yml [hunk 2]
```diff
@@ -0,0 +1,21 @@
+name: Linting Code
+
+on: [push]
```
Comment (gruselhaus) [inline .github/workflows/ci-lint.yml]:
```suggestion
on: [push, pull_request]
```

Same thing here.

File: .github/workflows/ci-test.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: Testing
+
+on: [push]
```
Comment (gruselhaus) [inline .github/workflows/ci-test.yml]:
```suggestion
on: [push, pull_request]
```

Same thing here.

**Comment by lmccart** [comment]
🎉

**Pull request merged**